.github/workflows/_get_torch_commit.yml

```diff
@@ -0,0 +1,32 @@
+name: get-torch-commit
+on:
+  workflow_call:
+    outputs:
+      torch_commit:
+        description: "torch commit to be used"
+        value: ${{ jobs.get-commit.outputs.torch_commit }}
+
+jobs:
+  get-commit:
+    runs-on: ubuntu-20.04
+    outputs: 
+      torch_commit: ${{ steps.get_torch_commit.outputs.torch_commit }}
+    steps:
+      # See https://github.com/actions/checkout/issues/1014#issuecomment-1906802802
+      - name: Clean up workspace
+        run: |
+          ls -la
+          rm -rvf ${GITHUB_WORKSPACE}/*
+      - name: Checkout PyTorch Repo
+        uses: actions/checkout@v4
+        with:
+          repository: pytorch/pytorch
+          path: pytorch
+          submodules: recursive
+      - id: get_torch_commit
+        name: Get torch commit
+        run: |
+          cd pytorch
+          torch_commit=$(git rev-parse HEAD) 
+          echo "torch_commit=$torch_commit" >> "$GITHUB_OUTPUT"
```

Comment:
I just learned about the [`git ls-remote`](https://www.git-scm.com/docs/git-ls-remote) command, which looks perfect here. `git ls-remote https://github.com/pytorch/pytorch.git HEAD | awk '{print $1}'` should work, and then you don't have to clone.

If this ends up being a single step and we don't re-use it, is it possible to actually put that directly into `build_and_test.yaml`?